google-api-go-generator/gen.go

```diff
@@ -495,6 +495,19 @@ func (a *API) apiBaseURL() string {
 	return resolveRelative(base, rel)
 }
 
+func (a *API) mtlsApiBaseURL() string {
+	var base, rel string
+	switch {
+	case *baseURL != "":
+		base, rel = *baseURL, a.doc.BasePath
+	case a.doc.MTLSRootURL != "":
+		base, rel = a.doc.MTLSRootURL, a.doc.ServicePath
+	default:
+		base, rel = *apisURL, a.doc.BasePath
```

Comment:
Why does this fall-back to the default base path?

It should probably return empty string if there is none defined in the discovery doc.

Reply:
mtlsAPIBaseURL mirrors the logic in apiBaseURL, which uses the following fall-back order:
1. Check for baseURL flag override (this will be the authoritative endpoint, mtls or not, if provided)
2. If MTLSRootURL is available in discovery doc, use it.
3. Fall-back to apisURL, which is actually the discovery url "https://www.googleapis.com/discovery/v1/apis".
I'm not sure why step 3 returns discovery url instead of just empty. Either way, we should make the behavior consistent between mtls and nonMTLS. WDYT?

Reply:
not sure, that sounds like a bug. I think mTLS should read the field directly from the discovery doc and we can move any logic into the transport layer instead of having it in the generator.

Reply:
Ok, simplifying this to:
func (a *API) mtlsAPIBaseURL() string {
	if a.doc.MTLSRootURL != "" {
		return resolveRelative(a.doc.MTLSRootURL, a.doc.ServicePath)
	}
	return ""
}

Reply:
I think we will still need this fallback logic though, if the url is not in the discovery doc, the transport package should fallback to the auto constructed path

Reply:
I think if the mTLS url is not in the discovery doc (which would be rare), then in all likelihood the service does not support mTLS anyway (and the enable_mtls flag would probably be false in the service config to begin with). In a subsequently PR, I will add the logic to incorporate the "enable_mtls" flag into the client, which will authoritatively control whether to allow mTLS endpoint in the client. See go/mtls-client-side-whitelisting.